### PR TITLE
Clean Makefile

### PR DESCRIPTION
**Pull request by rpellerin**
- Get rid of all the coffee files and keep generated JavaScript files
- Make the appropriate changes to Makefile
- Fix the `Assertion failed` that occurred with the message `Assertion failed: you need to wait for the runtime to be ready (e.g. wait for main() to be called)`
- Fix https://github.com/kripken/emscripten/issues/4990

### CODE REVIEW

**Comment by dinedal** [comment]
Seems like it's failing in travis ci?

**Comment by rpellerin** [comment]
Yes sorry, there's a variable scope problem. I'll fix this tomorrow morning.

**Comment by rpellerin** [comment]
Currently `SQL` isn't exposed directly, only through `Module`. I'll restore the old behavior tomorrow, i.e. make it  exposed globally.

**Comment by lovasoa** [comment]
Switching from coffeescript to JavaScript if not necessarily a bad idea, but if you want to do so, then you have to rewrite the API code as idiomatic js, you can't leave the generated files like that

**Comment by rpellerin** [comment]
From https://travis-ci.org/kripken/sql.js/builds/207117119, one test passes, on Node 0.10, the others fail at Reading Blob.

**Comment by lovasoa** [comment]
Travis also indicates a warning message in the tests

> writeStringToMemory is deprecated and should not be called! Use stringToUTF8() instead!

File: js/api.js [hunk 1]
```diff
@@ -1,16 +1,54 @@
...
+SQLite.TEXT = 3;
+SQLite.BLOB = 4;
+SQLite.NULL = 5;
+SQLite.UTF8 = 1;
+
+var SQL = {}
 
 Module['onRuntimeInitialized'] = function() {
-  // Generated by CoffeeScript 1.12.4
+    console.log('ok')
```
Comment (lovasoa) [inline js/api.js]: Why did you add a `console.log` ? Libraries should have undocumented side-effects, like writing to the console.

File: js/api.js [hunk 2]
```diff
@@ -0,0 +1,780 @@
...
+
+    Statement.prototype.getBlob = function(pos) {
+      var i, k, ptr, ref, result, size;
+      if (pos == null) {
+        pos = this.pos++;
+      }
+      size = sqlite3_column_bytes(this.stmt, pos);
+      ptr = sqlite3_column_blob(this.stmt, pos);
+      result = new Uint8Array(size);
+      for (i = k = 0, ref = size; 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
```
Comment (lovasoa) [inline js/api.js]: This is unreadable.

File: js/api.js [hunk 3]
```diff
@@ -0,0 +1,780 @@
...
+        while (stmt.step()) console.log(stmt.get());
+     */
+
+    Statement.prototype['get'] = function(params) {
+      var field, k, ref, results1;
+      if (params != null) {
+        this['bind'](params) && this['step']();
+      }
+      results1 = [];
+      for (field = k = 0, ref = sqlite3_data_count(this.stmt); 0 <= ref ? k < ref : k > ref; field = 0 <= ref ? ++k : --k) {
```
Comment (lovasoa) [inline js/api.js]: This is unreadable.

File: js/api.js [hunk 4]
```diff
@@ -0,0 +1,780 @@
...
+
+        var stmt = db.prepare("SELECT 5 AS nbr, x'616200' AS data, NULL AS nothing;");
+        stmt.step(); // Execute the statement
+        console.log(stmt.getColumnNames()); // Will print ['nbr','data','nothing']
+     */
+
+    Statement.prototype['getColumnNames'] = function() {
+      var i, k, ref, results1;
+      results1 = [];
+      for (i = k = 0, ref = sqlite3_data_count(this.stmt); 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
```
Comment (lovasoa) [inline js/api.js]: This is unreadable.

File: js/api.js [hunk 5]
```diff
@@ -0,0 +1,780 @@
...
+
+    /* Bind values to numbered parameters
+    @param [Array]
+    @private
+    @nodoc
+     */
+
+    Statement.prototype.bindFromArray = function(values) {
+      var k, len, num, value;
+      for (num = k = 0, len = values.length; k < len; num = ++k) {
```
Comment (lovasoa) [inline js/api.js]: This is not idiomatic.

File: js/api.js [hunk 6]
```diff
@@ -0,0 +1,780 @@
...
+        stmt.step(); // Execute the statement
+        console.log(stmt.getAsObject()); // Will print {nbr:5, data: Uint8Array([1,2,3]), nothing:null}
+     */
+
+    Statement.prototype['getAsObject'] = function(params) {
+      var i, k, len, name, names, rowObject, values;
+      values = this['get'](params);
+      names = this['getColumnNames']();
+      rowObject = {};
+      for (i = k = 0, len = names.length; k < len; i = ++k) {
```
Comment (lovasoa) [inline js/api.js]: This is not idiomatic.

File: js/api.js [hunk 7]
```diff
@@ -0,0 +1,780 @@
...
+
+
+    /* Exports the contents of the database to a binary array
+    @return [Uint8Array] An array of bytes of the SQLite3 database file
+     */
+
+    Database.prototype['export'] = function() {
+      var _, binaryDb, ref, stmt;
+      ref = this.statements;
+      for (_ in ref) {
```
Comment (lovasoa) [inline js/api.js]: This is not idiomatic.

File: js/api.js [hunk 8]
```diff
@@ -0,0 +1,780 @@
...
+    be used anymore.
+
+    Databases **must** be closed, when you're finished with them, or the
+    memory consumption will grow forever
+     */
+
+    Database.prototype['close'] = function() {
+      var _, ref, stmt;
+      ref = this.statements;
+      for (_ in ref) {
```
Comment (lovasoa) [inline js/api.js]: This is not idiomatic.

File: js/api.js [hunk 9]
```diff
@@ -0,0 +1,780 @@
...
+    @param name [String] the name of the function as referenced in SQL statements.
+    @param func [Function] the actual function to be executed.
+     */
+
+    Database.prototype['create_function'] = function(name, func) {
+      var func_ptr, wrapped_func;
+      wrapped_func = function(cx, argc, argv) {
+        var arg, args, data_func, i, k, ref, result, value_ptr, value_type;
+        args = [];
+        for (i = k = 0, ref = argc; 0 <= ref ? k <= ref : k >= ref; i = 0 <= ref ? ++k : --k) {
```
Comment (lovasoa) [inline js/api.js]: This is unreadable.

File: js/api.js [hunk 10]
```diff
@@ -0,0 +1,780 @@
...
+                return sqlite3_value_double;
+              case value_type !== 3:
+                return sqlite3_value_text;
+              case value_type !== 4:
+                return function(ptr) {
+                  var blob_arg, blob_ptr, j, l, ref1, size;
+                  size = sqlite3_value_bytes(ptr);
+                  blob_ptr = sqlite3_value_blob(ptr);
+                  blob_arg = new Uint8Array(size);
+                  for (j = l = 0, ref1 = size; 0 <= ref1 ? l < ref1 : l > ref1; j = 0 <= ref1 ? ++l : --l) {
```
Comment (lovasoa) [inline js/api.js]: unreadable

File: js/api.js (hunk 1 above)
Comment (rpellerin) [inline js/api.js]: Omission, sorry. Removed in latest commit.

**Comment by bhollis** [comment]
@rpellerin I'm guessing this would be easier to merge if all the different things you're fixing were separate PRs.

**Comment by femski** [comment]
Any update ? Can this be merged ?

**Comment by dinedal** [comment]
> Any update ? Can this be merged ?

The travis tests are failing still, either they need to be updated or they need to pass as is.

**Comment by kaizhu256** [comment]
can this pull-request be closed as a duplicate of #349 ?
